src/main/cs/Infinispan/HotRod/Config/AuthenticationConfigurationBuilder.cs

```diff
@@ -50,6 +50,7 @@ void setupCallback(Infinispan.HotRod.SWIGGen.SaslCallbackHandlerMap map)
 
         public AuthenticationConfigurationBuilder SetupCallback(System.Collections.Generic.IDictionary<int, AuthenticationStringCallback> map)
         {
+            callbackMap = map;
```

Comment:
The other pull request adds a SetupStringCallback and that method doesn't store the map locally. Just something to remember.

Reply:
C++ rebuilds the callbacks map on its side, so should be fine to not keep the map as long as the single callback function are scoped by the application